wowchemy/assets/js/wowchemy-slides.js

```diff
@@ -0,0 +1,166 @@
+}
+
+pluginOptions["plugins"] = enabledPlugins;
+
+Reveal.initialize(pluginOptions);
+
+
+// mermaid not enabled by default
+if(params.slides.diagram === undefined){
+  params.slides.diagram = false
+}
+
+// configure mermaid only if enabled
+if (params.slides.diagram) {
+  //mermaid options
+  // mermaid: front matter configuration can be used to set mermaid options
+  // You can also use directives (see mermaid documentation)
+  var mermaidOptions = (typeof params.slides.diagram_options === 'undefined') ? {} : params.slides.diagram_options;
+
+  // startOnLoad must be false since diagrams are lazily rendered
+  mermaidOptions["startOnLoad"] = false;
+
+
+  mermaid.initialize(mermaidOptions);
+
+  // Following functions are used to render mermaid diagrams
+  // after reveal slides have been successfully loaded
+  // since content of slides is lazy loaded, if diagrams are
+  // rendered at start of presentation their sizes will be off
+  /* get all slides that are:
+1- data loaded
+2- display set to block
+3- has a mermaid element that is not processed (data-processed dne)
+
+*/
+  function mermaidSlidesReadyToRender(s) {
+    diag = s.querySelector(".mermaid");
+    if (diag) {
+      background = s.slideBackgroundElement;
+      // render if we are 1 slide away horizontally
+      // current visible slide index
+      chidx = Reveal.getState()["indexh"];
+
+      // mermaid slide index
+      shidx = Reveal.getIndices((slide = s))["h"];
+      if (
+        !diag.hasAttribute("data-processed") &&
+        background.hasAttribute("data-loaded") &&
+        background.style.display === "block" &&
+        shidx - chidx <= 1
+      )
+        return s;
+    }
+    return null;
+  }
+
+  function renderMermaidSlides() {
+    dSlides = Reveal.getSlides().filter(mermaidSlidesReadyToRender);
+    dSlides.forEach((item) => mermaid.init(item.querySelector(".mermaid")));
+  }
+
+  // render mermaid slides for slides that are ready
+  Reveal.on("slidechanged", (event) => {
+    renderMermaidSlides();
+  });
+
+  // render mermaid slides for slides that are ready on startup
+  Reveal.on("Ready", (event) => {
+    if (Reveal.isReady()) {
+      renderMermaidSlides();
+    }
+  });
+  // end of mermaid diagram functions
+
+  function fixMermaid() {
+    let mermaids = [];
+    [].push.apply(
+      mermaids,
+      document.getElementsByClassName("language-mermaid")
+    );
+    for (let i = 0; i < mermaids.length; i++) {
+      $(mermaids[i]).unwrap("pre"); // Remove <pre> wrapper.
+      $(mermaids[i]).replaceWith(function () {
+        // Convert <code> block to <div> and add `mermaid` class so that Mermaid will parse it.
+        return $("<div />").append($(this).contents()).addClass("mermaid");
+      });
+    }
+  }
+
+  {
+    {
+      /*  document.addEventListener("DOMContentLoaded", function(event) { 
+    fixMermaid();
+  });  */
+    }
+  }
+
+  // document ready
+  $(document).ready(function () {
```

Comment:
Please use vanilla JS rather than jQuery

Reply:
alright .. but the fixmermkid is still there and needs extraction